### PR TITLE
Call of Shitcurity: Nanite Horrors part 1 - Mystery Crate

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Adds a mystery crate item for antags, for 10tc.
It has 3 uses and will give you a random gun based on a weighted possibility list. Each use costs 10k credits.
The possibilities are weighed mostly on mediocre guns, but you could get something horrendously OP... or something just horrendously horrendous.

https://cdn.discordapp.com/attachments/640744619302453316/734994866081366116/2020-07-21_01-45-22.mp4

(it has sound effects i just didn't remember to record them)

## Why It's Good For The Game

This will be the first in a long series of cod zombies inspired PRs, and is fittingly one of the most notable icons of the series.

## Changelog
:cl:
add: Added mystery crate to the uplink for 10tc. Fetch me their souls.
/:cl:

### CODE REVIEW

File: modular_skyrat/code/modules/tidezombies/machinery/mysterycrate.dm [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	new randumb(T)
+	if(!uses)
+		QDEL_IN(src, 6 SECONDS)
+		playsound(src, byebye, 100, 0)
+
+/obj/structure/closet/crate/mysterycrate/Destroy()
+	. = ..()
+	new /obj/item/toy/plush/mysterycrate(src.loc)
+
+/obj/item/toy/plush/mysterycrate
```
Comment (CameronWoof) [inline modular_skyrat/code/modules/tidezombies/machinery/mysterycrate.dm:183]: ?

Comment (BobJoga) [reply modular_skyrat/code/modules/tidezombies/machinery/mysterycrate.dm:183]: the plush gets left behind when the crate gets deleted that is the meme

Comment (CameronWoof) [reply modular_skyrat/code/modules/tidezombies/machinery/mysterycrate.dm:183]: Is this someone's OC?

**Comment by CameronWoof** [comment]
My concern with this is mainly that by purchasing money from the upload, you're guaranteed to get all three uses out of this whenever you want, and you're very nearly guaranteed to get your money's worth. 16tc for three usable guns and a chance to get something especially powerful seems like a good deal for someone who's goal is simply to be hyperlethal.

Not sure if adding another murderbox is necessarily the best idea, though I do really like the core concept of something that potentially causes you to amass money from the crew for a concrete benefit.

**Comment by BobJoga** [comment]
> My concern with this is mainly that by purchasing money from the upload, you're guaranteed to get all three uses out of this whenever you want, and you're very nearly guaranteed to get your money's worth. 16tc for three usable guns and a chance to get something especially powerful seems like a good deal for someone who's goal is simply to be hyperlethal.
> 
> Not sure if adding another murderbox is necessarily the best idea, though I do really like the core concept of something that potentially causes you to amass money from the crew for a concrete benefit.

I can rig the weights of guns to be more shitty if that is such a concern. That said, it's not a murderbox because most of the time you get something mediocre, it's more of a literal lootbox just like surplus crates - you could get godly or get shitted.

**Comment by Gandalf2k15** [comment]
Love this idea, will codecheck later.

**Comment by Gandalf2k15** [comment]
Maybe make it consume previous guns if it doesn't already(not code dived it as I'm sleeping)

**Comment by BobJoga** [comment]
> Maybe make it consume previous guns if it doesn't already(not code dived it as I'm sleeping)

It doesn't, the intent is for it to have many uses but it costs a shit ton for each

(currently 3 uses, might change if it proves to be too unbalanced)

**Comment by Investigator77** [comment]
Could comment out the plushie and use the one in which I provided the regular lizard

**Comment by BobJoga** [comment]
> Could comment out the plushie and use the one in which I provided the regular lizard

i did that no worries

**Comment by IrkallaEpsilon** [comment]
/obj/item/gun/ballistic/shotgun = 50, //you know why

Because it doesnt miss?

**Comment by Investigator77** [comment]
What's wrong with Nuke Op and Pulse weapons when you can literally get them from gateway missions?

**Comment by KathrinBailey** [comment]
> What's wrong with Nuke Op and Pulse weapons when you can literally get them from gateway missions?

Those missions are, or at least should be, disabled pending the removal of powergamer loot and (more likely) a full rework.
Upstream also supposedly removed a bunch of them. If they actually did or not is anyone's guess.

**Comment by Investigator77** [comment]
Well for one, you can still get microbombs, a Unica, L6 SAW, S.H.H. HMG, Pulse pistol, Desert Eagle, 1911 and CTF Mag-fed laser rifle in the gateway. And those are the only ones I remember.

On another note, leaving out both pulse and nukeop guns puts the risk-reward of having a 40×32 crate that eats up 10k credits per use isn't very beneficial. If you remove those, you can only end up with half-goof stuff like M1911s (Literally a stechkin with an extra round) and the regular .357 revolver. If a crate doesn't have a chance to roll something unique for the price, then it's not really worth it at all.

On that note, the surplus crate can come with Nuke Op items too.

**Comment by CameronWoof** [comment]
Kathrin _just_ addressed that gateway loot is problematic and undergoing review.

As for the rest... if the best this crate offers is already offered by another crate (which can contain nonlethal, gimmick, or otherwise more interesting items than guns), is that not a good indication it might not be a good addition in its current state?

**Comment by BobJoga** [comment]
> Properly document your changes.
> image
> This is the sort of thing that belongs in another PR. Said PR would get merged extremely fast. If you want it in this PR you have to at least actually mention the change and preferably do it in the changelog too.
> 
> Remove:
> Pickle gun
> All magic wands
> All pulse weapons
> All Nukeops weapons
> Meteor gun.
> 
> At that point an actual discussion can be had on this and its place, if it has one.
> I'm already very concerned that you decided to put admin weapons in here. Given that you put their weight to 1 it's obvious you know what they are.

I'm removing the plushie, about everything else:
I literally don't know what to say if you're missing the point of this item so fucking bad, Kat. There is literally no possible reason for this to exist if you can't try your luck to get something horrendously OP. At that point, i will just buy a surplus crate, which will get me way more of my money's worth. Their weight are set to 1 for that exact reason, so i don't know what point you're trying to make. Do you want me to bump it to 10, so everyone can have a pickle gun? I'm not removing the fucking nuke ops weapons and everything else because it's the only reason anyone would ever buy this.

**Comment by BobJoga** [comment]
> Kathrin _just_ addressed that gateway loot is problematic and undergoing review.
> 
> As for the rest... if the best this crate offers is already offered by another crate (which can contain nonlethal, gimmick, or otherwise more interesting items than guns), is that not a good indication it might not be a good addition in its current state?

It's not a good addition for your boring ass, not everything has to be fucking stealthy you dickwad. This is an interesting gimmick item for someone who just wants a fucking randomized gun without the possibility of getting 15 radio jammers instead.

**Pull request closed without merging**